### PR TITLE
#10 - [WARNING] Attempt to create a file for type 'package.to.MyEditor_field_Context' multiple times when running tests

### PR DESCRIPTION
**Pull request by FrankHossfeld**

### CODE REVIEW

File: gwt-editor-processor/src/test/resources/org/gwtproject/editor/processor/test08/TestEditor08.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  private Driver driver;
+  
+  @IsDriver
+  interface Driver
+      extends SimpleBeanEditorDriver<Model01Dto, TestEditor08> {
+    
+  }
+  
+  FakeLeafValueEditorWithHasEditorDelegate<Long> id;
+  FakeLeafValueEditorWithHasEditorDelegate<String>                    name;
```
Comment (niloc132) [inline gwt-editor-processor/src/test/resources/org/gwtproject/editor/processor/test08/TestEditor08.java:22]: extrrrrrrrrrra whitespace

File: gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java [hunk 2]
```diff
@@ -272,13 +281,35 @@ private ClassName getEditorDelegate(EditorModel editorModel, EditorProperty data
...
-        FieldSpec.builder(rawEditorType, "editor", Modifier.PRIVATE).build());
+    if (((DeclaredType) data.getEditorType()).getTypeArguments().size() == 2) {
+      delegateTypeBuilder.addField(
+          ParameterizedTypeName.get(
+              ClassName.get((TypeElement) types.asElement(data.getEditorType())),
+              ClassName.get(((DeclaredType) data.getEditorType()).getTypeArguments().get(0)),
+              ClassName.get(((DeclaredType) data.getEditorType()).getTypeArguments().get(1))),
+          "editor",
+          Modifier.PRIVATE);
+    } else if (((DeclaredType) data.getEditorType()).getTypeArguments().size() == 1) {
```
[the remark this answers is not in the document]
Comment (niloc132) [reply gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java:302]: Consider CompositeEditor as an example of a case with three args

File: gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java [hunk 3]
```diff
@@ -147,6 +148,14 @@ private void generate(TypeElement interfaceToImplement, EditorModel rootEditorMo
     String pkgName = this.elements.getPackageOf(interfaceToImplement).getQualifiedName().toString();
     String typeName = createNameFromEnclosedTypes(interfaceToImplement, "_Impl");
 
+    String classNameToGenerate = pkgName + "." + typeName;
+    // check, if tyoename is already generatd ...
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java:152]: fixed

File: gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java [hunk 4]
```diff
@@ -272,13 +281,35 @@ private ClassName getEditorDelegate(EditorModel editorModel, EditorProperty data
             //                      .addMember("value", "\"$L\"",
             // DriverProcessor.class.getCanonicalName())
             //                      .build())
-            .superclass(getEditorDelegateType()); // raw type here, for the same reason as above
+            .superclass(
+                ParameterizedTypeName.get(
+                    getEditorDelegateType(),
+                    ClassName.get(data.getEditedType()),
+                    ClassName.get(
+                        data.getEditorType()))); // raw type here, for the same reason as above
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java:289]: removed

File: gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java [hunk 5]
```diff
@@ -290,10 +321,17 @@ private ClassName getEditorDelegate(EditorModel editorModel, EditorProperty data
...
+            ParameterizedTypeName.get(
+                getEditorDelegateType(),
+                ClassName.get(d.getEditedType()),
+                ClassName.get(d.getEditorType())),
+            fieldName,
+            Modifier.PRIVATE);
       }
     }
 
+    //    asdsadasdasd
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java:333]: oh, my bookmark ... removed

File: gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java:302]: Oh, yes, looks like Script Kid Programming .... Think, found a better solution.

File: gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java [hunk 6]
```diff
@@ -514,6 +560,9 @@ private ClassName getEditorContext(EditorProperty parent, EditorProperty data) {
             .addModifiers(Modifier.PUBLIC)
             .addAnnotation(Override.class)
             .returns(Class.class)
+            //            .returns(
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java:563]: Deleted it. Will look into it with #11

File: gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java [hunk 7]
```diff
@@ -307,8 +345,8 @@ private ClassName getEditorDelegate(EditorModel editorModel, EditorProperty data
             .addModifiers(Modifier.PROTECTED)
             .returns(void.class)
             .addAnnotation(Override.class)
-            .addParameter(Editor.class, "editor")
-            .addStatement("this.editor = ($T) editor", rawEditorType)
+            .addParameter(ClassName.get(data.getEditorType()), "editor")
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java:348]: I moved back to the original implementation, cause we have another issue regarding the generation warning issue. (#11) Might be the better place to discuss it.

File: gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java [hunk 8]
```diff
@@ -324,8 +362,8 @@ private ClassName getEditorDelegate(EditorModel editorModel, EditorProperty data
             .addModifiers(Modifier.PROTECTED)
             .returns(void.class)
             .addAnnotation(Override.class)
-            .addParameter(ClassName.get(Object.class), "object")
-            .addStatement("this.object = ($T) object", ClassName.get(data.getEditedType()))
+            .addParameter(ClassName.get(data.getEditedType()), "object")
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-editor-processor/src/main/java/org/gwtproject/editor/processor/DriverProcessor.java:365]: see a answer above

File: gwt-editor-processor/pom.xml [hunk 9]
```diff
@@ -49,8 +49,8 @@
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
 
-        <auto.common.version>0.10</auto.common.version>
-        <auto.service.version>1.0-rc6</auto.service.version>
+        <auto.common.version>0.11</auto.common.version>
+        <auto.service.version>1.0-rc7</auto.service.version>
```
Comment (niloc132) [inline gwt-editor-processor/pom.xml:53]: 1.1 and 1.0 are latest of these now

File: gwt-editor-processor/src/test/resources/org/gwtproject/editor/processor/test08/result/TestEditor08_SimpleBeanEditorDelegate.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
+package org.gwtproject.editor.processor.test08;
+import java.lang.Object;
+import java.lang.Override;
+import org.gwtproject.editor.client.Editor;
+import org.gwtproject.editor.client.EditorVisitor;
+import org.gwtproject.editor.client.impl.SimpleBeanEditorDelegate;
+import org.gwtproject.editor.client.testing.FakeLeafValueEditorWithHasEditorDelegate_Long_SimpleBeanEditorDelegate;
+import org.gwtproject.editor.client.testing.FakeLeafValueEditorWithHasEditorDelegate_String_SimpleBeanEditorDelegate;
+import org.gwtproject.editor.processor.common.Model01Dto;
+public class TestEditor08_SimpleBeanEditorDelegate extends SimpleBeanEditorDelegate {
```
Comment (niloc132) [inline gwt-editor-processor/src/test/resources/org/gwtproject/editor/processor/test08/result/TestEditor08_SimpleBeanEditorDelegate.java:10]: looks like we're still missing generics here

**Review by niloc132 — COMMENTED** [review]
Let's fix the versions and ship

**Review by niloc132 — APPROVED** [review]

File: gwt-editor-processor/src/test/resources/org/gwtproject/editor/processor/test08/result/TestEditor08_SimpleBeanEditorDelegate.java (hunk 10 above)
Comment (FrankHossfeld) [reply gwt-editor-processor/src/test/resources/org/gwtproject/editor/processor/test08/result/TestEditor08_SimpleBeanEditorDelegate.java:10]: correct ... that is another upcoming PR

**Pull request merged**